File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var countTriplets = function(arr) {
    const n = arr.length + 1;
    let ans = 0;
    let prefix = Array(n).fill(0);

    for(let i = 1; i<n; i++) {
       prefix[i] = prefix[i-1] ^ arr[i-1];
    }

    for(let i = 0; i<n; i++) {
        for(let j = i+1; j<n; j++) {
            if(prefix[i] == prefix[j]) {
                ans += (j - i -1);
            }
        }
    }

    return ans;
};

File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        int n = arr.size(), ans = 0;
        vector<int> prefix(n+1, 0);

        for(int i = 1; i<n+1; i++) prefix[i] = prefix[i-1] ^ arr[i-1];

        for(int i = 0; i<n+1; i++) {
            for(int j = i + 1; j<n+1; j++) {
                if(prefix[i] == prefix[j]) {
                    ans += (j - i - 1);
                }
            }
        }

        return ans;
    }
};